shared/podman/images.go

```diff
@@ -6,48 +6,175 @@ package podman
 
 import (
 	"bytes"
+	"encoding/json"
 	"fmt"
+	"io"
+	"os"
+	"path"
 	"strings"
 
 	"github.com/rs/zerolog"
 	"github.com/rs/zerolog/log"
+	"github.com/uyuni-project/uyuni-tools/shared/types"
 	"github.com/uyuni-project/uyuni-tools/shared/utils"
 )
 
 // Ensure the container image is pulled or pull it if the pull policy allows it.
-func PrepareImage(image string, pullPolicy string, args ...string) error {
+func PrepareImage(image string, pullPolicy string, args ...string) (string, error) {
 	log.Info().Msgf("Ensure image %s is available", image)
 
-	needsPull, err := checkImage(image, pullPolicy)
+	present, err := IsImagePresent(image)
 	if err != nil {
-		return err
+		return image, err
 	}
 
-	if needsPull {
-		return pullImage(image, args...)
+	if present && strings.ToLower(pullPolicy) != "always" {
+		log.Debug().Msgf("Image %s already present", image)
+		return image, nil
+	}
+
+	rpmImageFile, err := GetRpmImagePath(image)
+	if err != nil {
+		return image, err
+	}
+
+	if len(rpmImageFile) > 0 {
+		log.Debug().Msgf("Image %s present as RPM. Loading it", image)
+		loadedImage, err := loadRpmImage(rpmImageFile)
+		if err != nil {
+			log.Warn().Msgf("cannot use RPM image for %s:%s", image, err)
+			present = false
+		} else {
+			if strings.ToLower(pullPolicy) == "always" {
+				log.Debug().Msg("Ignoring pull policy alway ")
```

Comment:
```suggestion
				log.Debug().Msg("Ignoring pull policy 'always'")
```
Should we really ignore the pull policy? I would just apply it: if the image is already loaded, just skip.

Reply:
ok, I was not sure about the best to do.

Reply:
I wouldn't ignore the policy: that would be too confusing for users